### PR TITLE
Allow SelectField and PhoneField to focus on mounting

### PR DESCRIPTION
**Pull request by nickoola**

### CODE REVIEW

**Comment by nickoola** [comment]
Capture d’écran 2019-07-08 à 10 35 58
Capture d’écran 2019-07-08 à 10 34 32
Capture d’écran 2019-07-08 à 10 33 31

**Comment by nickoola** [comment]
> Few high level questions:
> 
> * you added a autoFocus on selectField but not on the phoneField. I suppose the autofocus was added to autofocus something inside the phonefield. But I'm not sure how you re gonna use that select param if it's not propagated from phonefield
> * autofocus only works on page load AFAIK but you can access pages with a phonefield after a client-side navigation (and then autofocus will not work). Why not rely on a programmatic focus behavior that will work all the time and will be needed anyway for SPA navigation ?
> * For Phonefield, do we plan to have a focus style when either the nested select or textinput is selected inside the phonefield ? ( there is none for the moment IIRC)

I finally add `focus` and `autoFocus` for both components.
And I added attached screens to display the render on mounting `focus` is setted to `true`.

File: src/phoneField/index.tsx [hunk 1]
```diff
@@ -181,6 +186,20 @@ export default class PhoneField extends PureComponent<PhoneFieldProps, PhoneFiel
...
+    if (this.ref && this.props.focus) {
+      this.ref.current.focus()
+      this.setState({ hasFocus: true })
+    }
+  }
+
+  componentDidUpdate(prevProps: PhoneFieldProps) {
+    if (this.ref && prevProps.focus !== this.props.focus) {
+      this.ref.current.focus()
+      this.setState({ hasFocus: true })
```
Comment (wakooka) [inline src/phoneField/index.tsx]: Remove this line since it will trigger the `onFocus`

File: src/phoneField/index.tsx [hunk 2]
```diff
@@ -181,6 +186,20 @@ export default class PhoneField extends PureComponent<PhoneFieldProps, PhoneFiel
     return null
   }
 
+  componentDidMount() {
+    if (this.ref && this.props.focus) {
+      this.ref.current.focus()
+      this.setState({ hasFocus: true })
```
Comment (wakooka) [inline src/phoneField/index.tsx]: Remove this line since it will trigger the `onFocus`

File: src/phoneField/index.tsx (hunk 1 above)
Comment (nickoola) [reply src/phoneField/index.tsx]: Done!

File: src/phoneField/index.tsx (hunk 2 above)
Comment (nickoola) [reply src/phoneField/index.tsx]: Done!

**Pull request merged**